### ISSUE TITLE
is the log softmax function missing in the Transformer Example?

### ISSUE THREAD

**Issue by hilfe123** [pytorch/tutorials#977]
Hello,
The tutorial (https://pytorch.org/tutorials/beginner/transformer_tutorial.html) describes the Transformer paper and says that "... .To have the actual words, the output of nn.TransformerEncoder model is sent to the final Linear layer, which is followed by a log-Softmax function." The code, however, does return the output directly from the last Linear layer and does not use a (log) softmax anywhere. Do I fail to see something or is it actually missing?

**Comment by shamoons** [comment]
Same question - did you get an answer?

**Comment by hilfe123** [comment]
Hi, yes. The "solution" is the module that calculates the loss, which automatically applies the log softmax. So there is no need to code it again. From the docs: "This criterion combines nn.LogSoftmax() and nn.NLLLoss() in a single class (https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html). So, even though the output itself is strictly speaking not a LogSoftmax it is used to calculate the loss. If you want to use the model to output a distribution during inference you might want to use nn.LogSoftmax() explicitly.
